gix-dir/src/entry.rs

```diff
@@ -26,11 +26,13 @@ pub enum Property {
 /// The kind of the entry, seated in their kinds available on disk.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Ord, PartialOrd)]
 pub enum Kind {
-    /// Something that is not a file, like a named pipe or character device.
+    /// Something that is not a regular file, directory, or symbolic link.
     ///
-    /// These can only exist in the filesystem.
+    /// These can only exist in the filesystem, because Git repositories do not support them.
+    /// They do not appear as blobs in a repository, and their type is not specifiable in a tree object.
+    /// Examples include named pipes (FIFOs), character devices, block devices, and sockets.
     NonFile,
-    /// The entry is a blob, executable or not.
+    /// The entry is a blob, representing a regular file, executable or not.
```

Comment:
This is changing documentation of something other than `NonFile`s. My rationale is twofold:

1. Regular files are not the only kind of blob, because symlinks are also blobs:

   ```text
   120000 blob 965b606f331b51d566b46025f9ff311a7aad0c12    LICENSE-APACHE
   120000 blob 76219eb72e8524f15c21ec93b9b2592da49b5460    LICENSE-MIT
   ```

2. The documentation of `NonFile` now refers to not being representable as a blob, so avoiding implying that only regular files are representable as blobs seems like a benefit. In addition, in another files, `types.rs`, the documentation of `NonFile`s now explicitly distinguishes them from regular files, directories, and symlinks, so I think there's value in explicitly documenting the items here in a way that includes all three of those terms where they apply.

(Of those two points, only the second speaks to why this seems, to me, to be in scope for this PR.)